core/src/avm2/globals/avmplus.rs

```diff
@@ -80,6 +83,48 @@ bitflags::bitflags! {
     }
 }
 
+fn describe_type_json_null<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    flags: DescribeTypeFlags,
+) -> Result<Value<'gc>, Error<'gc>> {
+    // These checks will effectively cause `describeType` to faithfully mirror FP's
+    // implementation. That is to say, it will cause it to throw an "Error #1009: Cannot access
+    // a property or method of a null object reference."
+    if flags.contains(DescribeTypeFlags::USE_ITRAITS)
+        || !flags.contains(DescribeTypeFlags::INCLUDE_BASES | DescribeTypeFlags::INCLUDE_TRAITS)
```

Comment:
Where does this `!flags.contains(DescribeTypeFlags::INCLUDE_BASES | DescribeTypeFlags::INCLUDE_TRAITS)` come from? I haven't been able to get `describeTypeJSON` to return `null` except when `USE_ITRAITS` (the first condition) is set.

Reply:
By the way you can directly test the output of `describeTypeJSON` with a SWF that uses a function in the `avmplus` package to interface with `describeTypeJSON`, like so:
```
package avmplus
{
    public function describeTypeHack(a:*, b:*) : *
    {
        return describeTypeJSON(a,b);
    }
}
```